backend/src/utils/logtail.ts

```diff
@@ -1,29 +1,45 @@
 import { Logtail } from "@logtail/node";
 import type { Context, ILogtailLog } from "@logtail/types";
 
-const logtailInstance = new Logtail(String(process.env.LOGTAIL_TOKEN) || "");
+/**
+ * Logger class wraps Logtail logger. It will use Logtail on production and Console on development.
+ */
+export class Logger {
+  private readonly _logtailInstance: Console | Logtail;
 
-type WrapperHandler = (
-  message: string,
-  context: Context
-) => Promise<ILogtailLog & Context> | undefined;
-interface LogtailWrapper {
-  warn: WrapperHandler;
-  info: WrapperHandler;
-  error: WrapperHandler;
-}
+  /**
+   * @param token Logtail token. Should not be empty on production.
+   */
+  constructor(token: string | undefined) {
+    const isTokenEmpty = token === null || token === undefined || token === "";
+    const isProduction = process.env.NODE_ENV === "production";
+
+    if (isProduction && isTokenEmpty) {
+      throw TypeError("Logtail token should not be empty in production!");
+    }
+
+    this._logtailInstance =
+      isProduction && !isTokenEmpty ? new Logtail(token) : console;
+  }
 
-export const logtail: LogtailWrapper = {
-  warn(message, context = {}) {
-    if (process.env.NODE_ENV !== "production") return;
-    return logtailInstance.warn(message, context);
-  },
-  info(message, context = {}) {
-    if (process.env.NODE_ENV !== "production") return;
-    return logtailInstance.info(message, context);
-  },
-  error(message, context = {}) {
-    if (process.env.NODE_ENV !== "production") return;
-    return logtailInstance.error(message, context);
+  warn(
+    message: string,
+    context: Context = {}
+  ): Promise<ILogtailLog & Context> | void {
+    return this._logtailInstance.warn(message, context);
   }
-};
+  info(
+    message: string,
+    context: Context = {}
+  ): Promise<ILogtailLog & Context> | void {
+    return this._logtailInstance.info(message, context);
+  }
+  error(
+    message: string,
+    context: Context = {}
+  ): Promise<ILogtailLog & Context> | void {
+    return this._logtailInstance.error(message, context);
+  }
+}
+
+export const logger = new Logger(process.env.LOGTAIL_TOKEN);
```

Comment:
should provide default value for LOGTAIL_TOKEN

Reply:
ngga perlu kan? biar dia pakenya Console, bukan Logtail

Reply:
oh ok